lib/geordi/gitlinear.rb

```diff
@@ -0,0 +1,201 @@
+require 'highline'
+require 'net/http'
+require 'json'
+require 'active_support/core_ext/object/blank'
+
+module Geordi
+  class Gitlinear
+    # This require-style is to prevent Ruby from loading files of a different
+    # version of Geordi.
+    require File.expand_path('settings', __dir__)
+
+    API_ENDPOINT = 'https://api.linear.app/graphql'.freeze
+
+    def initialize
+      self.highline = HighLine.new
+      self.settings = Settings.new
+    end
+
+    def run_commit(git_args)
+      Interaction.warn <<~WARNING unless Util.staged_changes?
+        No staged changes. Will create an empty commit.
+      WARNING
+
+      issue = choose_issue
```

Comment:
Maybe we can fail here just like in L31? Would give more feedback to the user. And renders the `if` obsolete :)

Reply:
If you agree, we should move the `fail` into `choose_issue`. WDYT?

Reply:
Thanks for the suggestion! I moved the fail to `choose_issue`, that actually looks much cleaner.